flytekit/bin/entrypoint.py

```diff
@@ -222,98 +224,141 @@ def _handle_annotated_task(task_def: PythonTask, inputs: str, output_prefix: str
-                    # the environment variable. Look up the correct input/outputs in the index lookup mapping file.
-                    job_index = _map_job_index_to_child_index(input_dir, inputs, job_index)
+                # If an ArrayTask is discoverable, the original job index may be different than the one specified in
+                # the environment variable. Look up the correct input/outputs in the index lookup mapping file.
+                job_index = _map_job_index_to_child_index(input_dir, inputs, job_index)
 
-                    inputs = _os.path.join(inputs, str(job_index), "inputs.pb")
-                    output_prefix = _os.path.join(output_prefix, str(job_index))
+                inputs = _os.path.join(inputs, str(job_index), "inputs.pb")
+                output_prefix = _os.path.join(output_prefix, str(job_index))
 
-                _data_proxy.Data.get_data(inputs, local_inputs_file)
-                input_proto = _utils.load_proto_from_file(_literals_pb2.LiteralMap, local_inputs_file)
+            _data_proxy.Data.get_data(inputs, local_inputs_file)
+            input_proto = _utils.load_proto_from_file(_literals_pb2.LiteralMap, local_inputs_file)
 
-                _engine_loader.get_engine().get_task(task_def).execute(
-                    _literal_models.LiteralMap.from_flyte_idl(input_proto),
-                    context={"output_prefix": output_prefix, "raw_output_data_prefix": raw_output_data_prefix},
-                )
+            _engine_loader.get_engine().get_task(task_def).execute(
+                _literal_models.LiteralMap.from_flyte_idl(input_proto),
+                context={"output_prefix": output_prefix, "raw_output_data_prefix": raw_output_data_prefix},
+            )
 
-            # New core style task
-            elif not test and isinstance(task_def, PythonTask):
-                _handle_annotated_task(task_def, inputs, output_prefix, raw_output_data_prefix)
+
+def _load_resolver(resolver_location: str) -> TaskResolverMixin:
+    # Load the actual resolver - this cannot be a nested thing, whatever kind of resolver it is, it has to be loadable
+    # directly from importlib
+    # TODO: Handle corner cases, like where the first part is [] maybe
+    resolver = resolver_location.split(".")
+    resolver_mod = resolver[:-1]  # e.g. ['flytekit', 'core', 'python_auto_container']
+    resolver_key = resolver[-1]  # e.g. 'default_task_resolver'
+    resolver_mod = _importlib.import_module(".".join(resolver_mod))
+    return getattr(resolver_mod, resolver_key)
 
 
 @_scopes.system_entry_point
-def _execute_map_task(task_module, task_name, inputs, output_prefix, raw_output_data_prefix, max_concurrency, test):
-    task_module = _importlib.import_module(task_module)
-    task_def = getattr(task_module, task_name)
+def _execute_task(inputs, output_prefix, raw_output_data_prefix, test, resolver: str, resolver_args: List[str]):
```

Comment:
i think we need documentation here for each of these methods to explain why they exist